components/core/src/os/process/windows.rs

```diff
@@ -47,3 +50,147 @@ fn become_child_command(command: PathBuf, args: Vec<OsString>) -> Result<()> {
     // Let's honor the exit codes from the child process we finished running
     process::exit(status.code().unwrap())
 }
+
+fn handle_from_pid(pid: u32) -> Option<winapi::HANDLE> {
+    unsafe {
+        let proc_handle = kernel32::OpenProcess(
+            winapi::PROCESS_QUERY_LIMITED_INFORMATION,
+            winapi::FALSE,
+            pid as winapi::DWORD
+        );
+
+        // we expect this to happen if the process died
+        // before OpenProcess completes
+        if proc_handle == ptr::null_mut() {
+            return None;
+        }
+        else {
+            return Some(proc_handle);
+        }
+    }
+}
+
+pub struct Child {
+    handle: Option<winapi::HANDLE>,
+    last_status: Option<u32>,
+    pid: u32
+}
+
+impl Child {
+    // On windows we need the process handle to capture status
+    // Here we will attempt to get the handle from the pid but if the
+    // process dies before we can get it, we will just wait() on the
+    // std::process::Child and cache the exit_status which we will return 
+    // when ststus is called.
+    pub fn new(child: &mut process::Child) -> Child {
+        let (win_handle, status) = match handle_from_pid(child.id()) {
+            Some(handle) => (Some(handle), None),
+            _ => (None, Some(child.wait().unwrap().code().unwrap() as u32))
```

Comment:
We actually want to panic if it errors here because in that case we have we have lost contact with the process, but we should totally panic with a better error.